Lib9c/Helper/PetHelper.cs

```diff
@@ -7,13 +9,41 @@ public static class PetHelper
         public static (int ncgQuantity, int soulStoneQuantity) CalculateEnhancementCost(
             PetCostSheet costSheet,
             int petId,
-            int nowLevel,
+            int currentLevel,
             int targetLevel)
         {
-            var costList = costSheet[petId].Cost;
-            var range = targetLevel - nowLevel;
-            var startIndex = nowLevel != 0
-                ? costList.FindIndex(cost => cost.Level == nowLevel) + 1
+            if (costSheet is null)
+            {
+                throw new ArgumentNullException(nameof(costSheet));
+            }
+
+            if (petId < 0)
+            {
+                throw new ArgumentException(
+                    $"{nameof(petId)} must be greater than or equal to 0.");
+            }
+
+            if (currentLevel < 0)
+            {
+                throw new ArgumentException(
+                    $"{nameof(currentLevel)} must be greater than or equal to 0.");
+            }
+
+            if (targetLevel < 0)
+            {
+                throw new ArgumentException(
+                    $"{nameof(targetLevel)} must be greater than or equal to 0.");
+            }
+
+            if (!costSheet.TryGetValue(petId, out var row))
+            {
+                throw new SheetRowNotFoundException(nameof(PetCostSheet), petId);
+            }
```

Comment:
Thanks for handling edge cases.
Can you add one more case: `if (currentLevel >= targetLevel)` ?